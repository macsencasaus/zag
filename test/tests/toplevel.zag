// TEST: lex, ir, x86_64-linux

extern fn printf(fmt: *u8, ...) i32;

var x = 10;
var y = -5;
var z = 0;
var a = 123456;
var nums = []i32{1, 2, 3, 4, 5};
var empty = []i32{};
var moreNums = []i32{42, 100, -7};

fn main() i32 {
    // Simple integer prints
    printf(&"x = %d\n", x);
    printf(&"y = %d\n", y);
    printf(&"z = %d\n", z);
    printf(&"a = %d\n", a);

    // Accessing array values
    printf(&"nums[0] = %d\n", nums[0]);
    printf(&"nums[4] = %d\n", nums[4]);

    printf(&"moreNums[1] = %d\n", moreNums[1]);
    printf(&"moreNums[2] = %d\n", moreNums[2]);

    printf(&"empty list size = %d\n", 0);

    return 0;
}
