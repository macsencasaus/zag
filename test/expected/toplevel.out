x = 10
y = -5
z = 0
a = 123456
nums[0] = 1
nums[4] = 5
moreNums[1] = 100
moreNums[2] = -7
empty list size = 0
